peps/tests/test_converters.py

```diff
@@ -43,6 +48,41 @@ def test_add_image_not_found(self):
             r"Image Path '(.*)/path/that/does/not/exist(.*)' does not exist, skipping"
         )
 
+    def test_add_image_update(self):
+        """Test that image modifications are adopted. """
+        pep = '3001'
+        pep_img_name = 'pep-3001-1.png'
+        pep_img_path_src = os.path.join(FAKE_PEP_REPO, pep_img_name)
+        pep_img_path_src_backup = pep_img_path_src + ".backup"
+        pep_img_path_dst = os.path.join(
+                settings.MEDIA_ROOT, pep_url(pep), pep_img_name)
+
+        # We need to remove a prior uploaded file to fix an inconsistency
+        # between db and filesystem.
+        # TODO: This should be dealt with in page.model or converters module
```

Comment:
Here's some additional info for this line;

A [prior test that calls `add_pep_image`](https://github.com/python/pythondotorg/blob/bc91a39d97a54943205d901c80c8a86b5bc30dde/peps/tests/test_commands.py#L36) might have already created the 'pep-3001-1.png' file, without persisting the db object. Now a new call to `add_pep_image` will [check for objects](https://github.com/python/pythondotorg/blob/bc91a39d97a54943205d901c80c8a86b5bc30dde/peps/converters.py#L217), whose [name ends with 'pep-3001-1.png'](https://github.com/python/pythondotorg/blob/bc91a39d97a54943205d901c80c8a86b5bc30dde/peps/converters.py#L221), and, since it does not find any, create it anew along with the file. But since Django does not allow to override files it duplicates it using a random infix, e.g 'pep-3001-1_0u7tlLY.png', which is also used as a name on the object.

Now due to the `endswith`-check and with the random infix, each subsequent attempt to add 'pep-3001-1.png' with `add_pep_image`, will also duplicate it the object and file.

Maybe we should just override images on upload as suggested as alternative fix in the PR description?